### ISSUE TITLE
Choices functionality breaking with images

### ISSUE THREAD

**Issue by dexius-ram-depop** [sgl-project/sglang#408]
When I am asking the model about an image, behaviour seems to break when I use the _choices_ functionality, and it seems to always suggest the first option. 
I think there is also an issue with normalisation when images are involved
### Setting up the server
`pip install outlines==0.0.34`
`pip install rpyc`
`pip install ./sglang/python`
`pip install vllm=0.3.3`
`pip install transformers==4.37.2`
Although I notice that the exact same issues exists on the version of sglang that is on pyPI
`python -m sglang.launch_server --model-path "liuhaotian/llava-v1.5-7b" --tokenizer-path "llava-hf/llava-1.5-7b-hf" --port 8082`
### Calling the model
```import requests
import torch
from PIL import Image
from io import BytesIO
import sglang as sgl

def load_image_from_url(image_path: str) -> Image:
  """Load image from url"""
  response = requests.get(image_path)
  return Image.open(BytesIO(response.content), mode='r')

@sgl.function
def image_qa_unrestricted(s, image):
    try:
      s += sgl.user("What animal is in the following image? " + sgl.image(image)) 
      s += sgl.assistant("a " + sgl.gen("response"))
    except:
      print(image)
@sgl.function
def image_qa_restricted(s, image, options):
    try:
      s += "What animal is in the following image? " + sgl.image(image) 
      s += "a " + sgl.gen("response",choices=options)
    except:
      print(image)
```
![cat](https://t4.ftcdn.net/jpg/00/97/58/97/360_F_97589769_t45CqXyzjz0KXwoBZT9PRaWGHRk5hQqQ.jpg)
```
image_path='https://t4.ftcdn.net/jpg/00/97/58/97/360_F_97589769_t45CqXyzjz0KXwoBZT9PRaWGHRk5hQqQ.jpg'
loaded_image = load_image_from_url(image_path).resize((500,500))

sgl.set_default_backend(sgl.RuntimeEndpoint("http://localhost:8082"))

options = ['shark','dog','cat']
options_2 = ['dog','cat','shark']

state_unrestricted = image_qa_unrestricted.run(loaded_image,temperature=0.0)
state_restricted = image_qa_restricted.run(loaded_image, options, temperature= 0.0)
state_restricted_2 = image_qa_restricted.run(loaded_image, options_2, temperature = 0.0)

meta_info_unrestricted = state_unrestricted.get_meta_info('response')
meta_info_restricted = state_restricted.get_meta_info('response')
meta_info_restricted_2 = state_restricted_2.get_meta_info('response')

print("Unrestricted model response: ", state_unrestricted['response'])
print("Restricted model response (shark/dog/cat): ", state_restricted['response'])
print("Restricted model response (dog/cat/shark): ", state_restricted_2['response'])

print("Log probabilities (shark/dog/cat): ", meta_info_restricted['normalized_prompt_logprobs'])
print("Log probabilities (dog/cat/shark): ",meta_info_restricted_2['normalized_prompt_logprobs'])
```
### Raw output
```
Unrestricted model response:   cat with green eyes.
Restricted model response (shark/dog/cat):  shark
Restricted model response (dog/cat/shark):  dog
Log probabilities (shark/dog/cat):  [-5.934884548187256, -13.6631498336792, -13.658100128173828]
Log probabilities (dog/cat/shark):  [-5.9516282081604, -13.658093452453613, -13.648798942565918]
```
I don't think that permuting the choices should affect the probabilities in this way (it should instead permute the probabilities), and also the fact that the values are so similar I think is due to the fact that if you run
`print(meta_info_restricted["prefill_token_logprobs"][0]` you get a response that is of length ~500, and I think when normalising it divides over the length of the input. So even though the image is not being generated, it is still affecting the normalisation
